src/Url.php

```diff
@@ -99,8 +95,10 @@ private static function split($url)
                 $url = '';
             }
         } else {
-            $scheme = '';
-            $domain = '';
+            throw new InvalidArgumentException(sprintf(
```

Comment:
Could you invert the if condition and move this to the if-block? Then you can get rid of the else-block.